### PR TITLE
scripts: update to latest west

### PR DESCRIPTION
**Pull request by mbolivar**
Now that the multi-repo West functionality is ready for wider testing, we need to tweak the launcher scripts so that they can detect whether the zephyr tree is part of a multi-repo installation, and run the west code in its own repository if that is the case.

Do the needful, preserving compatibility with monorepo installations.

### CODE REVIEW

**Review by carlescufi — APPROVED** [review]
Tested locally

**Comment by carlescufi** [comment]
recheck

**Review by carlescufi — CHANGES_REQUESTED** [review]
NACKing until we sort a recently discovered issue

File: scripts/west-launcher.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    sys.path.append(os.path.join(os.environ['ZEPHYR_BASE'], 'scripts', 'meta'))
+
+    from west.main import main as west_main  # noqa E402 (silence flake8)
+
+    west_main(sys.argv[1:])
+
+
+def main():
+    try:
+        topdir = find_west_topdir(os.getcwd())
```
Comment (mbolivar) [inline scripts/west-launcher.py]: This os.getcwd() should be __file__. I'm also working on emitting a message when this script is being run from a "monorepo" Zephyr (i.e. a zephyr tree outside of a multi-repo west installation) with the west bootstrapper installed on the system.

**Comment by mbolivar** [comment]
> NACKing until we sort a recently discovered issue

Ready for retesting

**Comment by mbolivar** [comment]
> Ready for retesting

@carlescufi --- but as discussed, we can't merge unless https://github.com/zephyrproject-rtos/west/pull/50 is OK as-is, since that's where this west came from. If west 50 needs changes, this will too.

**Comment by mbolivar** [comment]
(I see the docs now fail to build; looking into it)

**Review by carlescufi — APPROVED** [review]
Retested locally

**Comment by carlescufi** [comment]
@mbolivar there's a doc issue:
```
WARNING: autodoc: failed to import module 'west.runners.core' the following exception was raised:
No module named 'runners'
```

**Pull request merged**